aibolit/patterns/send_null/send_null.py

```diff
@@ -9,59 +9,28 @@ class SendNull:
     def __init__(self):
         pass
 
-    def __find_position_recursively(self, node):
-        if not hasattr(node, 'children'):
-            return
-        else:
-            for i in node.children:
-                if not isinstance(i, list):
-                    if hasattr(i, '_position'):
-                        if i.position:
-                            return i._position.line
-                        else:
-                            for j in i.children:
-                                position = self.__find_position_recursively(j)
-                                if position:
-                                    return position
-                else:
-                    for j in i:
-                        position = self.__find_position_recursively(j)
-
-        return position
-
-    # flake8: noqa
-    # after fix addition, errors are shown
     def value(self, filename: str) -> List[int]:
 
-        lst: List[int] = []
-        tree = build_ast(filename)
-
-        invocation_tree = tree.filter(javalang.tree.Invocation)
-        arg_list = [x for _, x in invocation_tree]
-
-        for argument in arg_list:
-            ternary_list = argument.filter(javalang.tree.TernaryExpression)
-            for _, expr in ternary_list:
-                if isinstance(expr.if_false, javalang.tree.Literal) and expr.if_false.value == 'null':
-                    if hasattr(argument, '_position'):
-                        lst.append(argument._position.line)
-                    else:
-                        position = self.__find_position_recursively(expr)
-                        lst.append(position)
-                if isinstance(expr.if_true, javalang.tree.Literal) and expr.if_true.value == 'null':
-                    if hasattr(argument, '_position'):
-                        lst.append(argument._position.line)
-                    else:
-                        position = self.__find_position_recursively(expr)
-                        lst.append(position)
-
-        for _, node in tree:
-            try:
-                for argument in node.arguments:
-                    if isinstance(argument, javalang.tree.Literal) and argument.value == "null" and \
-                            argument._position.line not in lst:
-                        lst.append(argument._position.line)
-            except (AttributeError, TypeError):
-                pass
-        lst = sorted(lst)
+        lines = set()
+        tree = AST.build_from_javalang(build_ast(filename))
+        invocatios_types = [
+            ASTNodeType.METHOD_INVOCATION,
+            ASTNodeType.EXPLICIT_CONSTRUCTOR_INVOCATION,
+            ASTNodeType.CLASS_CREATOR
+        ]
+        for node in tree.get_proxy_nodes(*invocatios_types):
+            for argument in node.arguments:
+                if (argument.node_type == ASTNodeType.LITERAL) and (argument.value == "null"):
```

Comment:
Maybe just omit parentheses?

Reply:
If you really need... In future if you add smth else, you can make a mistake. It is more readable

Reply:
Ok then